### DISCUSSION TITLE
Questions about img2img

### DISCUSSION THREAD

**Discussion by Abel1011** [vpzomtrrfrt/stability-client#6] [Q&A]
Hi, thanks for the library. It works very well so far. I am trying to add the img2img option in my application and reviewing the code I see that you include these parameters:

imagePrompt: {
        mime: string;
        content: Buffer;
    } | null;
    stepSchedule: {
        start?: number;
        end?: number;
    };

Can these values be used to process img2img? What does the stepSchedule represent or what does it do? In the web version of dreamstudio there is a parameter "streng", is it possible to include it in the library?

**Comment by vpzomtrrfrt** [comment]
Yes, these are used for img2img. The API doesn't have any concept of "strength", but it seems to be related to the step schedule. I don't actually know what the values mean

**Reply by Abel1011** [reply]
Hi, in Stability's Phyton sdk there is initImage and maskImage; in this case, how can I use the mask Image?

**Reply by vpzomtrrfrt** [reply]
just released 1.7.0 with support for setting the mask

**Reply by Abel1011** [reply]
Thank you very much for the good work. One additional query, is it possible in the API response, to know when an image will be blurred.

**Reply by vpzomtrrfrt** [reply]
yes, the image data includes `classifications.realizedAction`, which will be non-zero if the result has tripped the filter

**Comment by sillytuna** [comment]
From a quick test, this would be 35% Image Strength.

```
let imageStrength = 0.5;

//...
stepSchedule: {
	start: imageStrength,
	end: 1 - imageStrength
},

```
